src/memefactory/server/syncer.cljs

```diff
@@ -310,45 +310,60 @@
        (log/info (str "Blacklisting address " address) ::apply-blacklist-patches)
        (db/patch-forbidden-reg-entry-image! address)))))
 
+(defn- block-timestamp* [block-number]
+  (js/Promise.
+   (fn [resolve reject]
+     (web3-eth/get-block @web3 block-number false (fn [err {:keys [:timestamp] :as res}]
+                                                    (if err
+                                                      (reject err)
+                                                      (do
+                                                        (log/debug "cache miss for block-timestamp" {:block-number block-number :timestamp timestamp})
+                                                        (resolve timestamp))))))))
+
+(def block-timestamp
+  (memoize block-timestamp*))
```

Comment:
I'm not sure if memoize function is gonna actually work with promises like this

Reply:
why not? It just builds a cache, what this cache returns is secondary. Just try it ;)

Reply:
hmm, how is what it returns secondary? 
https://clojuredocs.org/clojure.core/memoize
> The memoized version of the function keeps a cache of the mapping from arguments
to results

Reply:
Secondary in the sense that the result is data, any data, a number, a function, a js/object etc.

Reply:
ah, so it returns the same Promise object, that's basically already resolved?

Reply:
Voila, i actually suspect that beneath its just building a hashmap